Stage 2

zurabreon opened the pull request:
Начало работы над 2-ым заданием

zurabreon committed:
feat

Реализовал подключение к mongoDB

zurabreon committed:
feat

Выполнил второе задание

aivanov-reon commented on authSync.ts:
```diff
@@ -1,20 +1,24 @@
+const DB_NAME = 'account-collection';
+const ACCOUNT_COLLECTION = 'accounts';
+
+const db = 'mongodb+srv://zguysanovreon:Qwerty321123@cluster0.usmz3it.mongodb.net/';
```
Почему используешь кластер? Можно локально поставить и использовать локальную базу данных.

zurabreon replied:
Поставил локальную базу данных

aivanov-reon reviewed (commented):
Я уверен, что твой код работает и скорее всего даже правильно, но нужно разносить логику.
Мы в проектах используем пакет не mongodb а mongoose - используй его
Чтобы сделать красиво и хорошо, нужно следующее сделать: 

1. Создай отдельный модуль для подключения к БД, DataBaseClientService  там будет метод для подключения
2.  На страте приложения создай инстанс клиента и сделай подключения к БД
3. Нужно создать схему и модель данных для аккаунта
4. Создать сервис для управления данными аккаунта AccountService, в сервис  внедрить модель через нее делать запросы к БД, написать методы для добавления, обновления, удаления. 
5.  Применить методы после получения токена и его рефреша или при удалении аккаунта

По итогу у тебя логика будет разнесена по разным логическим модулям и authSync не будет зависеть от базы данных

zurabreon committed:
feat

Реализованы отдельный модуль подключения бд, сервис управления данными аккаунта,

zurabreon committed:
refactor

aivanov-reon commented on services/DataBaseClientService.ts:
```diff
@@ -0,0 +1,30 @@
+
+}
+
+const connectDB = async () => {
```
В класс + добавить типы возвращаемых значений

zurabreon replied:
Обернул в класс, добавил возвращаемые типы

aivanov-reon commented on services/AccountService.ts:
```diff
@@ -0,0 +1,67 @@
+import { AccountModel } from "../models/accountModel"
+import { connectDB, disconnectDB } from "./DataBaseClientService";
+
+const addAccount = async (account_id: String, domain: String, access_token: String, refresh_token: String, installed: Boolean) => {
```
В класс + добавить типы возвращаемых значений

zurabreon replied:
Обернул в класс, добавил возвращаемые типы. Есть вопрос с addAccount

aivanov-reon commented on index.ts:
```diff
@@ -42,4 +42,4 @@ app.get('/logout', async (req: Request<unknown, unknown, unknown, RequestQuery>,
 
 
-app.listen(config.PORT,() => mainLogger.debug('Server started on ', config.PORT));
+app.listen(config.PORT,() => mainLogger.debug('Server started on ', config.PORT));
```
Организовать подключение к БД тут

zurabreon replied:
Организовал подключение бд в listen

aivanov-reon commented on authSync.ts:
```diff
@@ -1,20 +1,18 @@
 axiosRetry(axios, { retries: 3, retryDelay: axiosRetry.exponentialDelay });
 
-
 class AuthSync extends Api {
```
Внедрить сервис в качестве зависимости и использовать его тут

zurabreon replied:
Внедрил класс в качестве зависимости

zurabreon committed:
refactor

Сделал правки

zurabreon committed:
refactor

Добавил отдельный тип для аккаунта

zurabreon committed:
refactor
